server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/http/UserCredentialParserTest.scala

```diff
@@ -0,0 +1,144 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.http
+
+import java.nio.charset.StandardCharsets
+import java.util.Base64
+
+import eu.timepit.refined.auto._
+import org.apache.james.core.Username
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class UserCredentialParserTest {
+  @Test
+  def shouldReturnCredentialsWhenUsernamePasswordToken(): Unit = {
+    val token: String = "Basic " + toBase64("user1:password")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(Some(UserCredential(Username.of("user1"), "password")))
+  }
+
+  @Test
+  def shouldAcceptPartSeparatorAsPartOfPassword(): Unit = {
+    val token: String = "Basic " + toBase64("user1:pass:word")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(Some(UserCredential(Username.of("user1"), "pass:word")))
+  }
+
+  @Test
+  def shouldReturnCredentialsWhenRandomSpecialCharacterInUsernameToken(): Unit = {
+    val token: String = "Basic " + toBase64("fd2*#jk:password")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(Some(UserCredential(Username.of("fd2*#jk"), "password")))
+  }
+
+  @Test
+  def shouldReturnCredentialsWhenRandomSpecialCharacterInBothUsernamePasswoedToken(): Unit = {
+    val token: String = "Basic " + toBase64("fd2*#jk:password@fd23*&^$%")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(Some(UserCredential(Username.of("fd2*#jk"), "password@fd23*&^$%")))
+  }
+
+  @Test
+  def shouldReturnCredentialsWhenUsernameDomainPasswordToken(): Unit = {
+    val token: String = "Basic " + toBase64("user1@domain.tld:password")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(Some(UserCredential(Username.of("user1@domain.tld"), "password")))
+  }
+
+  @Test
+  def shouldReturnCredentialsWhenUsernameDomainNoPasswordToken(): Unit = {
+    val token: String = "Basic " + toBase64("user1@domain.tld:")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(Some(UserCredential(Username.of("user1@domain.tld"), "")))
+  }
+
+  @Test
+  def shouldReturnNoneWhenPayloadIsNotBase64(): Unit = {
+    val token: String = "Basic user1:password"
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(None)
+  }
+
+  @Test
+  def shouldThrowWhenInvalidToken(): Unit = {
+    val token: String = "Basic " + toBase64("user1@password")
+
+    assertThat(UserCredential.parseUserCredentials(token))
+      .isEqualTo(None)
+  }
+
+  @Test
+  def shouldReturnEmptyWhenWithEmptyToken(): Unit = {
+    assertThat(UserCredential.parseUserCredentials(""))
+      .isEqualTo(None)
+  }
+
+  @Test
+  def shouldThrowWhenWithWrongFormatToken(): Unit = {
```

Comment:
It looks to be the same test as `shouldThrowWhenInvalidToken` no?

Reply:
They are the same, we can remove `shouldThrowWhenInvalidToken`